### DISCUSSION TITLE
Strategie de branching

### DISCUSSION THREAD

**Discussion by mis7ern9o** [DstMlOpsCrypto/MainCrypto#10] [General]
Hello all,

La strategie de branching étant à definir (Trunk base ou GitFlow) nous devons statuer sur ce point.
Voici un petit résumé ci-apres mais pour plus d'informations merci de vous rendre sur ce lien [trunk-based-development-vs-git-flow](https://blog.mergify.com/trunk-based-development-vs-git-flow-when-to-use-which-development-style/)

---

# STRATEGIE DE BRANCHING (resumé)

## Strategies
Deux stratégies co existent : GitFlow base et Trunk base

### Strategie GitFlow base

- **principe :**
  - utilise des branches de fonctionnalité et plusieurs branches primaires.
  - les développeurs créent une branche de fonctionnalité et attendent que la fonctionnalité soit terminée pour la merger à la branche « trunk » principale

- **avantages :** 
  - adapté aux projets avec un cycle de livraison planifié
  - assigne des rôles très spécifiques aux différentes branches

- **inconvenients :** 
  - Les branches de fonctionnalités "longues" sont plus susceptibles d'introduire des mises à jour conflictuelles (deviation branche de trunk)

- **note :** 
  - structure = main, release, develop, feature, hotfix
<img src="https://wac-cdn.atlassian.com/dam/jcr:34c86360-8dea-4be4-92f7-6597d4d5bfae/02%20Feature%20branches.svg" alt="Your image title" width="500"/>

### Strategie Trunk base

- **principe :**
  - dès qu'un dev termine une feature, celle-ci est mergé sur la branche principale
- **avantages :** 
  - adaptée pour la CI (flux régulier de commits)
  - assure une revue de code en continue et rapide (petits commits)
  - permet des livraisons fréquentes (PR quotidiens). Le code dans la branche main doit pouvoir être déployé en production)
- **inconvenients :** 
  - necessite des tests automatisés + build avant de merger sur la branche principale
- **note :** 
  - structure = branche principale (main) + branche temporaire (label = ticket feature git)

**Comment by FredericLAGNIEZ** [comment]
Pour un projet "simple" et une petite équipe comme la nôtre, mon avis est qu'il faut plutôt aller vers la Stratégie Trunk base; Cela nous impose que personne ne travaille sur la même branche, ce qui réduira les conflits de fusion. La stratégie est plus rapide et plus simple à mon sens. Je connais moins l'autre pratique.

**Comment by TWR-RWT** [comment]
Merci Yann. Je partage l'avis de Frederic, j'ai l'impression que la Stratégie Trunk base est la plus adaptée des deux.

**Comment by mis7ern9o** [comment]
je suis ok avec vous, c'est parti.
je vais donc bloquer les commits directes sur la branch main donc, seules les PR (pull request seront acceptées)
j'ai activé le code review, 2 users doivent approver (hors celui qui fait la PR bien sur), ceci pour etre au fait de ce qui se passe et permettre un double check ou autre
